File: javascript/08-jealous-function/src/kitchen_ticket.js
class Order {
  constructor(items, customer, table, special, rush = false) {
    this.items = items;
    this.customer = customer;
    this.table = table;
    this.special = special;
    this.rush = rush;
  }
}

class KitchenTicket {
  print_ticket(order) {
    let lines = [];
    lines.push('Table: ' + order.table.number);
    lines.push('Zone: ' + (order.table.zone || 'main'));
    lines.push('Server: ' + (order.table.server || 'unassigned'));
    lines.push('Customer: ' + order.customer.name);
    if (order.customer.vip) lines.push('VIP');
    if (order.rush) lines.push('RUSH');
    let totalItems = 0;
    for (let i = 0; i < order.items.length; i++) {
      totalItems += order.items[i].qty;
      let line = order.items[i].name + ' x' + order.items[i].qty;
      if (order.items[i].modifiers && order.items[i].modifiers.length > 0) {
        line += ' [' + order.items[i].modifiers.join(', ') + ']';
      }
      if (order.items[i].allergy) {
        line += ' ALLERGY:' + order.items[i].allergy;
      }
      lines.push(line);
    }
    lines.push('Items: ' + totalItems);
    if (order.special && order.special.length > 0) {
      lines.push('Special: ' + order.special);
    }
    lines.push('---');
    return lines.join('\n');
  }
}

export { KitchenTicket, Order };

File: csharp/08-jealous-function/src/kitchen_ticket.cs
public class TicketItem
{
    public string Name;
    public int Qty;
    public List<string> Modifiers;
    public string Allergy;

    public TicketItem(string name, int qty, List<string>? modifiers = null, string allergy = "")
    {
        Name = name;
        Qty = qty;
        Modifiers = modifiers ?? new List<string>();
        Allergy = allergy;
    }
}

public class KitchenOrder
{
    public List<TicketItem> Items;
    public Dictionary<string, object> Customer;
    public Dictionary<string, object> Table;
    public string Special;
    public bool Rush;

    public KitchenOrder(List<TicketItem> items, Dictionary<string, object> customer, Dictionary<string, object> table, string special, bool rush = false)
    {
        Items = items;
        Customer = customer;
        Table = table;
        Special = special;
        Rush = rush;
    }
}

public class KitchenTicket
{
    public string print_ticket(KitchenOrder order)
    {
        var lines = new List<string>();
        lines.Add("Table: " + order.Table["number"]);
        lines.Add("Zone: " + (order.Table.ContainsKey("zone") ? order.Table["zone"] : "main"));
        lines.Add("Server: " + (order.Table.ContainsKey("server") ? order.Table["server"] : "unassigned"));
        lines.Add("Customer: " + order.Customer["name"]);
        if (order.Customer.ContainsKey("vip") && (bool)order.Customer["vip"])
        {
            lines.Add("VIP");
        }
        if (order.Rush)
        {
            lines.Add("RUSH");
        }
        var totalItems = 0;
        foreach (var item in order.Items)
        {
            totalItems += item.Qty;
            var line = item.Name + " x" + item.Qty;
            if (item.Modifiers.Count > 0)
            {
                line += " [" + string.Join(", ", item.Modifiers) + "]";
            }
            if (!string.IsNullOrEmpty(item.Allergy))
            {
                line += " ALLERGY:" + item.Allergy;
            }
            lines.Add(line);
        }
        lines.Add("Items: " + totalItems);
        if (!string.IsNullOrEmpty(order.Special))
        {
            lines.Add("Special: " + order.Special);
        }
        lines.Add("---");
        return string.Join("\n", lines);
    }
}
